File: cc/dvec2.cc
#include <string>
#include <iostream>
#include <vector>
#include <stdio.h>
#include <stdlib.h>

using namespace std;

string f(unsigned long n)
{
  vector<string> v;

  char buffer[32];

  for (unsigned long c = 0; c < n; ++c) {
    snprintf(buffer, 32, "%lu", c); // convert integer to string
    v.push_back(string(buffer) + "x");
  }

  return v[(size_t)rand() % v.size()];
}

int main(int argc, char **argv) {
  long n = atol(argv[1]);
  long r = atol(argv[2]);
  vector<string> v;

  for (unsigned long c = 0; c < r; ++c) {
    v.push_back(f(n));
  }

  cout << v[0] << endl;
}

File: ruby/dvec2.rb
def f(n)
  v = []
  for c in 0 ... n
    v << "#{c}x"
  end
  v.sample
end

if __FILE__ == $0
  n = Integer(ARGV[0])
  r = Integer(ARGV[1])
  v = []
  for c in 0 ... r
    v << f(n)
  end
  p v[0]
end
